File: atinject-core/src/main/java/org/atinject/api/registration/dto/IsUsernameAvailableRequest.java
package org.atinject.api.registration.dto;

import org.atinject.core.websocket.dto.WebSocketRequest;

public class IsUsernameAvailableRequest extends WebSocketRequest {

    private static final long serialVersionUID = 1L;
    
    private String username;
    
    public IsUsernameAvailableRequest(){
        super();
    }

    public String getUsername() {
        return username;
    }

    public IsUsernameAvailableRequest setUsername(String username) {
        this.username = username;
        return this;
    }
    
}

File: atinject-integration/src/main/resources/webapp/scripts/atinject/registration/IsUsernameAvailableRequest.js
define(["atinject/WebSocketRequest"],
function(WebSocketRequest){
    function IsUsernameAvailableRequest() {
        this._class = "org.atinject.api.registration.dto.IsUsernameAvailableRequest"
    };

    IsUsernameAvailableRequest.prototype = new WebSocketRequest();

        IsUsernameAvailableRequest.prototype.getUsername = function() {
        return this.username;
    };

    IsUsernameAvailableRequest.prototype.setUsername = function(username) {
        this.username = username;
        return this;
    };


    
    return IsUsernameAvailableRequest;
});
